src/mumble/UserLocalNicknameDialog.cpp

```diff
@@ -0,0 +1,118 @@
+   are met:
+
+   - Redistributions of source code must retain the above copyright notice,
+	 this list of conditions and the following disclaimer.
+   - Redistributions in binary form must reproduce the above copyright notice,
+	 this list of conditions and the following disclaimer in the documentation
+	 and/or other materials provided with the distribution.
+   - Neither the name of the Mumble Developers nor the names of its
+	 contributors may be used to endorse or promote products derived from this
+	 software without specific prior written permission.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+   ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+   A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR
+   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+   EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+   PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+   PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+   LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+   NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+   SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+
+#include "UserLocalNicknameDialog.h"
+#include "ClientUser.h"
+#include "Database.h"
+#include "MainWindow.h"
+
+#include <QtGui/QCloseEvent>
+#include <QtWidgets/QPushButton>
+
+// We define a global macro called 'g'. This can lead to issues when included code uses 'g' as a type or parameter name
+// (like protobuf 3.7 does). As such, for now, we have to make this our last include.
+#include "Global.h"
+
+UserLocalNicknameDialog::UserLocalNicknameDialog(unsigned int sessionId,
+											 QMap< unsigned int, UserLocalNicknameDialog * > *qmUserNicknameTracker)
+	: QDialog(nullptr), m_clientSession(sessionId), m_qmUserNicknameTracker(qmUserNicknameTracker) {
+	setupUi(this);
+	qleUserLocalNickname->setAccessibleName(tr("User nickname"));
+
+	ClientUser *user = ClientUser::get(sessionId);
+	if (user) {
+		QString title = tr("Adjusting local nickname for %1").arg(user->qsName);
+		setWindowTitle(title);
+		qleUserLocalNickname->setText(user->getLocalNickname());
+		m_originalNickname = qleUserLocalNickname->text();
+	}
+
+	if (g.mw && g.mw->windowFlags() & Qt::WindowStaysOnTopHint) {
+		// If the main window is set to always be on top of other windows, we should make the
+		// nickname dialog behave the same in order for it to not get hidden behind the main window.
+		setWindowFlags(Qt::WindowStaysOnTopHint);
+	}
+}
+
+void UserLocalNicknameDialog::closeEvent(QCloseEvent *event) {
+	m_qmUserNicknameTracker->remove(m_clientSession);
+	event->accept();
+}
+
+void UserLocalNicknameDialog::present(unsigned int sessionId,
+									QMap< unsigned int, UserLocalNicknameDialog * > *qmUserNicknameTracker) {

```

Comment:
Is it actually valid to pass `nullptr` as `qmUserNicknameTracker`? If not, then this should be a reference instead of a pointer :point_up:

Reply:
I would assume it's valid. That's how local volume adjustment currently uses qmUserVolTracker. If it's not valid, then we would have to change that implementation too.

Reply:
I just checked the code in more details and now I can say that passing `nullptr` here is not valid (will cause a segmentation fault as it is being dereferenced without a check for null).

The concept of using pointers only in cases in which `nullptr` is a valid case is unfortunately not really incorporated into the code base yet. Thus it can easily be that the volume adjustment dialog has the same issue.
If you feel like it, it'd be nice if you could also fix that in a separate `REFAC` commit. But if not, that's okay as well - just try to use this best practice in your new code :)